packages/flow-core/src/components/f-countdown/f-countdown.ts

```diff
@@ -0,0 +1,263 @@
+import { FRoot } from "../../mixins/components/f-root/f-root";
+import { FDiv } from "../f-div/f-div";
+import { flowElement } from "../../utils";
+import { injectCss } from "@cldcvr/flow-core-config";
+import getCustomFillColor from "../../utils/get-custom-fill-color";
+import { ref, createRef, Ref } from "lit/directives/ref.js";
+import { FText } from "../f-text/f-text";
+
+injectCss("f-countdown", globalStyle);
+
+const states = ["primary", "danger", "warning", "success", "default"] as const;
+const sizes = ["large", "medium", "small", "x-small"] as const;
+const categories = ["fill", "outline"] as const;
+const placements = ["left", "right", "bottom", "top", "none"] as const;
+
+export type FCountdownStateProp = (typeof states)[number];
+export type FCountdownCategoryProp = (typeof categories)[number];
+export type FCountdownSizesProp = (typeof sizes)[number];
+export type FCountdownLabelProp = (typeof placements)[number];
+
+@flowElement("f-countdown")
+export class FCountdown extends FRoot {
+	/**
+	 * css loaded from scss file
+	 */
+	static styles = [unsafeCSS(eleStyle), unsafeCSS(globalStyle), ...FDiv.styles, ...FText.styles];
+
+	/**
+	 * @attribute toggle accordion
+	 */
+	@property({ reflect: true, type: String })
+	category?: FCountdownCategoryProp = "fill";
+
+	/**
+	 * @attribute toggle accordion
+	 */
+	@property({ reflect: true, type: String })
+	duration?: number | string = 5;
+
+	/**
+	 * @attribute Each variant has various sizes. By default medium is the default size.
+	 */
+	@property({ type: String, reflect: true })
+	size?: FCountdownSizesProp = "medium";
+
+	/**
+	 * @attribute The states on tags are to indicate various degrees of emphasis of the action.
+	 */
+	@property({ reflect: true, type: String })
+	state?: FCountdownStateProp = "default";
+
+	/**
+	 * @attribute The states on tags are to indicate various degrees of emphasis of the action.
+	 */
+	@property({ reflect: true, type: String, attribute: "label-placement" })
+	labelPlacement?: FCountdownLabelProp = "left";
+
+	fill = "";
+
+	remaining = 0;
+
+	timerText = this.convertSecondsToMinutesAndSeconds(this.secondsDuration);
+
+	timerRef: Ref<FText> = createRef();
+
+	currentProgress = 0;
+
+	/**
+	 * apply inline styles to shadow-dom for custom fill.
+	 */
+	get applyStyles() {
+		if (this.fill) {
+			return `background: ${this.fill};`;
+		}
+		return "";
+	}
+
+	get applyPieStyles() {
+		return `background: var(--color-input-default); animation-duration: ${this.secondsDuration}s`;
+	}
+
+	get maskStyles() {
+		return `animation-duration: ${this.secondsDuration}s`;
+	}
+
+	get outlineTimerStyle() {
+		if (this.fill) {
+			return `--duration: ${this.secondsDuration}; --state-color:${this.fill}`;
+		}
+		return `--duration: ${this.secondsDuration}; --state-color:${
+			this.state !== "default"
+				? `var(--color-${this.state}-default)`
+				: `var(--color-primary-default)`
+		};`;
+	}
+
+	get secondsDuration() {
```

Comment:
Can we add more validations here, So we accept a number here which is greater than 0 and the upper limit needs to be 3600

if the string is specified then it must through an error for an invalid format